pre_commit/commands/autoupdate.py

```diff
@@ -33,6 +33,20 @@ class RevInfo(NamedTuple):
     def from_config(cls, config: dict[str, Any]) -> RevInfo:
         return cls(config['repo'], config['rev'], None)
 
+    def get_best_candidate_tag(self, rev: str, git_repo: str) -> str:
+        """Get the best tag candidate.
+
+        Multiple tags can exist on a SHA. Sometimes a moving tag is attached
+        to a version tag. Try to pick the tag that looks like a version.
+        """
+        tags = cmd_output(
+            'git', *git.NO_FS_MONITOR, 'tag', '--points-at', rev, cwd=git_repo,
+        )[1].split()
```

Comment:
I think `splitlines()` is slightly more correct -- though I can't find a whitespace character that's allowed in a tag